### PR TITLE
common : use common_ prefix for common library functions

### PR DESCRIPTION
**Pull request by slaren**
Prefix common library functions with `common_` instead of `llama_` to make it clear that these functions do not belong to the base API.

### CODE REVIEW

**Comment by slaren** [comment]
There are probably other functions that should be renamed as well - this was just a quick pass over `common.h` to see if there is interest in this change. I will do a more thorough pass if there is.

**Comment by ggerganov** [comment]
Yes, this is a good change.

**Comment by slaren** [comment]
Should I also rename the `gpt_` functions to `common_`?

**Comment by ggerganov** [comment]
Yes, along with the structs and enums (e.g. `struct gpt_params`, `enum gpt_sampler_type`, `gpt_log`).

File: common/common.h [hunk 1]
```diff
@@ -404,29 +404,29 @@ std::string fs_get_cache_file(const std::string & filename);
...
+    std::vector<common_lora_adapter_container> lora_adapters;
 };
 
-struct llama_init_result    llama_init_from_gpt_params(gpt_params & params);
+struct common_init_result     common_init_from_common_params(common_params & params);
 
-struct llama_model_params     llama_model_params_from_gpt_params    (const gpt_params & params);
-struct llama_context_params   llama_context_params_from_gpt_params  (const gpt_params & params);
+struct llama_model_params     common_model_params_from_common_params  (const common_params & params);
+struct llama_context_params   common_context_params_from_common_params(const common_params & params);
```
[the remark this answers is not in the document]
Comment (ggerganov) [reply common/common.h:416]: `common_init_from_common_params` -> `common_init_from_params`

Maybe:

`common_model_params_from_common_params` -> `common_model_params_to_llama`
`common_context_params_from_common_params` -> `common_context_params_to_llama`

**Review by ggerganov — APPROVED** [review]

**Pull request merged**